java/ingest/ingest-runner/src/main/java/sleeper/ingest/runner/impl/partitionfilewriter/S3TransferManagerWrapper.java

```diff
@@ -0,0 +1,132 @@
+import software.amazon.awssdk.transfer.s3.S3TransferManager;
+
+import sleeper.core.properties.instance.InstanceProperties;
+
+import java.net.URI;
+import java.util.Locale;
+
+import static sleeper.core.properties.instance.AsyncIngestPartitionFileWriterProperty.ASYNC_INGEST_CLIENT_TYPE;
+import static sleeper.core.properties.instance.AsyncIngestPartitionFileWriterProperty.ASYNC_INGEST_CRT_PART_SIZE_BYTES;
+import static sleeper.core.properties.instance.AsyncIngestPartitionFileWriterProperty.ASYNC_INGEST_CRT_TARGET_THROUGHPUT_GBPS;
+
+public class S3TransferManagerWrapper implements AutoCloseable {
+
+    private final S3TransferManager s3TransferManager;
+    private final S3AsyncClient closeS3AsyncClient;
+
+    private S3TransferManagerWrapper(S3TransferManager s3TransferManager, S3AsyncClient s3AsyncClient) {
+        this.s3TransferManager = s3TransferManager;
+        this.closeS3AsyncClient = s3AsyncClient;
+    }
+
+    public S3TransferManager get() {
+        return s3TransferManager;
+    }
+
+    @Override
+    public void close() {
+        if (closeS3AsyncClient != null) {
+            closeS3AsyncClient.close();
+        }
+        s3TransferManager.close();
+    }
+
+    public static S3TransferManagerWrapper create() {
+        return new S3TransferManagerWrapper(S3TransferManager.create(), null);
+    }
+
+    public static S3TransferManagerWrapper fromClient(S3AsyncClient s3AsyncClient) {
+        return new S3TransferManagerWrapper(S3TransferManager.builder().s3Client(s3AsyncClient).build(), null);
+    }
+
+    public static S3TransferManagerWrapper wrap(S3TransferManager s3TransferManager) {
+        return new S3TransferManagerWrapper(s3TransferManager, null);
+    }
+
+    public static S3TransferManagerWrapper s3AsyncClientOrDefaultFromProperties(
+            S3AsyncClient s3AsyncClient, InstanceProperties properties) {
+        if (s3AsyncClient == null) {
+            return fromProperties(properties);
+        } else {
+            return fromClient(s3AsyncClient);
+        }
+    }
+
+    public static S3TransferManagerWrapper fromProperties(InstanceProperties properties) {
+        S3AsyncClient s3AsyncClient = s3AsyncClientFromProperties(properties);
+        return new S3TransferManagerWrapper(S3TransferManager.builder().s3Client(s3AsyncClient).build(), s3AsyncClient);
+    }
+
+    public static S3AsyncClient s3AsyncClientFromProperties(InstanceProperties properties) {
+        String clientType = properties.get(ASYNC_INGEST_CLIENT_TYPE).toLowerCase(Locale.ROOT);
+        if ("java".equals(clientType)) {
+            return buildS3Client(S3AsyncClient.builder());
+        } else if ("crt".equals(clientType)) {
+            return buildCrtClient(S3AsyncClient.crtBuilder()
+                    .minimumPartSizeInBytes(properties.getLong(ASYNC_INGEST_CRT_PART_SIZE_BYTES))
+                    .targetThroughputInGbps(properties.getDouble(ASYNC_INGEST_CRT_TARGET_THROUGHPUT_GBPS)));
```

Comment:
This method is using instance properties specific to async ingest, but it isn't just used for async ingest now. I suppose we should clarify the method names and stop using this for the sketches store.